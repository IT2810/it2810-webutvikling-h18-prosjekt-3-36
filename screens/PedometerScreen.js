import Expo from "expo";
import React from "react";
import { Pedometer } from "expo";
import { connect } from "react-redux";
import { StyleSheet, Text, View } from "react-native";
import ProgressBar from "../components/ProgressBar";
import HeaderButton from "../components/HeaderButton";

class PedometerScreen extends React.Component {

  static navigationOptions = ({ navigation }) => {
    return {
      headerTitle: "Today's progress",
      headerRight: <HeaderButton onPress={() => navigation.navigate("PedometerSettings")} title="Edit goal" />
    };
  };
  state = {
    stepCount: 0,
  };

  // Kjør _refresh nå PedometerScreen lastes
  componentDidMount() {
    // Expo.Google.logInAsync(androidStandaloneAppClientId ="Ligger på Gdoccen.")
    this._refresh()
  }

  _refresh = () => {
    const end = new Date();
    end.setHours(23)
    end.setMinutes(59)
    end.setSeconds(59)

    const start = new Date();
    start.setHours(0)
    start.setMinutes(0)
    start.setSeconds(0)

    // Henter steps fra startdatoen til sluttdatoen og setter staten
    // Gir en feilmelding hvis det oppstår en error 
    Pedometer.getStepCountAsync(start, end).then(
      result => {
        this.setState({ stepCount: result.steps });
      },
      error => {
        this.setState({
          stepCount: "Could not get past step count: " + error
        });
      }
    );
    
    // 5 sekunder etter man har fått forrige resultat kjøres _refresh() på nytt
    setTimeout(() => {this._refresh()}, 5000);
  };

  render() {
    const today = this.state.stepCount;
    const {goal} = this.props;
    const percentage = Math.round(Math.min(today/goal * 100, 100));

    return (
      <View style={styles.container}>
        <Text style={styles.textHeader}>
          {percentage === 100 ? "🏆" : "🏃‍"} 
        </Text>

        <ProgressBar percentage={percentage} />
        <Text style={styles.textSmall}>
          {percentage + "% completed of today' s goal"}
        </Text>
        <Text style={styles.text}>
          Steps today: {"\n"}{this.state.stepCount}
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 30,
    paddingTop: 50,
    backgroundColor: 'white'
  },
  textHeader: {
    fontSize: 70,
    textAlign: 'center'
  },
  textSmall: {
    fontSize: 14,
    textAlign: 'center'
  },
  text: {
    color: 'black',
    textAlign: 'center',
    fontSize: 30,
    padding: 30,
  }
});

export default connect(state => ({ goal: state.settings.goal }))(PedometerScreen)

Expo.registerRootComponent(PedometerScreen);
