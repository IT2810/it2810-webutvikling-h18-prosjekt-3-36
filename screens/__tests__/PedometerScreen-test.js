import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import PedometerScreen from "../PedometerScreen";


it("PedometerScreen Matches snapshot.", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<PedometerScreen />)
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
});