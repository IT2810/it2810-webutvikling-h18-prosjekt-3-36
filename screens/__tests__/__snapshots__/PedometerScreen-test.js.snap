// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PedometerScreen Matches snapshot. 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 30,
      "paddingTop": 50,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 70,
        "textAlign": "center",
      }
    }
  >
    🏃‍
  </Text>
  <ProgressBar
    percentage={0}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 14,
        "textAlign": "center",
      }
    }
  >
    0% completed of today' s goal
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "fontSize": 30,
        "padding": 30,
        "textAlign": "center",
      }
    }
  >
    Steps today: 
    

    0
  </Text>
</Component>
`;
